parso/tree.py

```diff
@@ -266,9 +375,11 @@ def __init__(self, children: List[NodeOrLeaf]) -> None:
         """
         self.parent: Optional[BaseNode] = None
         '''
-        The parent :class:`BaseNode` of this leaf.
+        The parent :class:`BaseNode` of this node.
         None if this is the root node.
         '''
+        for child in children:
+            child.parent = self
```

Comment:
I don't like this, because this is already done by the parser and would therefore be done twice. The problem is that this is done a lot and it would probably slow down the parser for no good reason :).

If we want this (which is fine) we should probably remove it in the parser so it is not done twice. If you run into bugs it's probably the diff parser. That thing is a beast and if you run into problems I'm happy to help you there.

Reply:
Great, I think it looks good now. Thanks for changing it! I hope the diff parser does not fall flat on its face, but we'll find out :)